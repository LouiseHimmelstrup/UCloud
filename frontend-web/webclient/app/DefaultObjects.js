import {Cloud} from "../authentication/SDUCloudObject"
import {tusConfig} from "./Configurations";
import Uppy from "uppy";

export const DefaultStatus = {
    title: "No Issues",
    level: "NO ISSUES",
    body: "The system is running as intended.",
};

export const RightsMap = {
    NONE: 0,
    READ: 1,
    READ_WRITE: 2,
    OWN: 3,
};

export const AnalysesStatusMap = {
    "PENDING": 0,
    "IN PROGRESS": 1,
    "COMPLETED": 2,
};

export const RightsNameMap = {
    NONE: "None",
    READ: "Read",
    READ_WRITE: "Read/Write",
    OWN: "Own",
};

export const SensitivityLevel = {
    "OPEN_ACCESS": "Open Access",
    "CONFIDENTIAL": "Confidential",
    "SENSITIVE": "Sensitive",
};

export const SensitivityLevelMap = {
    "OPEN_ACCESS": 0,
    "CONFIDENTIAL": 1,
    "SENSITIVE": 2,
};

const initializeUppy = (restrictions) => 
    Uppy.Core({
        autoProceed: false,
        debug: false,
        restrictions: restrictions,
        meta: {
            sensitive: false,
        },
        onBeforeUpload: () => {
            return Cloud.receiveAccessTokenOrRefreshIt().then((data) => {
                tusConfig.headers["Authorization"] = `Bearer ${data}`;
            });
        }
    }).use(Uppy.Tus, tusConfig);

export const initObject = {
    files: {
        files: [],
        filesShown: 10,
        filesPerPage: 10,
        currentFilesPage: 0,
        loading: false,
        path: "",
        filesInfoPath: "",
        filesInfoFile: null,
        projects: []
    },
    uppy: {
        uppyFiles: initializeUppy({maxNumberOfFiles: false}),
        uppyFilesOpen: false,
        uppyRunApp: initializeUppy({ maxNumberOfFiles: 1}),
        uppyRunAppOpen: false
    },
    status: {
        status: DefaultStatus,
        title: ""
    }
};