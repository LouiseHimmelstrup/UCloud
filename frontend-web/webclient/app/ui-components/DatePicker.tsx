import ReactDatePicker, {ReactDatePickerProps} from "react-datepicker";
import "react-datepicker/dist/react-datepicker.css";
import theme from "@/ui-components/theme";
import {injectStyle} from "@/Unstyled";
import * as React from "react";
import Box, {BoxProps} from "@/ui-components/Box";

export const DatePickerClass = injectStyle("date-picker", k => `
    ${k} {
        appearance: none;
        display: block;
        font-family: inherit;
        color: inherit;
        font-size: ${theme.fontSizes[1]}px;
        background-color: transparent;
        border-radius: ${theme.radius};
        border-width: 0;
        border-style: solid;
        border-color: var(--borderGray, #f00);
        padding: 8px 12px;
        margin: 0;
        width: 100%;
        font-size: 18px;
    }
    
    ${k}::placeholder {
        color: var(--gray, #f00);
    }
    
    ${k}::-ms-clear {
        display: none;
    }
    
    ${k}:disabled {
        background-color: var(--lightGray, #f00);
    }
`);

export const DatePicker: React.FunctionComponent<ReactDatePickerProps<never, true> & BoxProps> = props => {
    return <Box {...props}><ReactDatePicker {...props} className={DatePickerClass} /></Box>;
};

DatePicker.displayName = "DatePicker";

export const SlimDatePickerClass = injectStyle("slim-date-picker", k => `
    ${k} .react-datepicker__day-name, ${k} .react-datepicker__day, ${k} .react-datepicker__time-name {
        width: 1.4rem;
        line-height: 1.4rem;
    }  
`);
