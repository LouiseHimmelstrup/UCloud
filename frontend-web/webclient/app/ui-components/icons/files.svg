<svg viewBox="0 0 24 22" xmlns="http://www.w3.org/2000/svg" fill-rule="evenodd" clip-rule="evenodd" stroke-linejoin="round" stroke-miterlimit="1.414"><path d="M0 21.315c0 .378.27.688.6.688h22.8c.33 0 .6-.31.6-.688v-16.5c0-.378-.27-.687-.6-.687H10.8L7.2.003H.6c-.33 0-.6.309-.6.687v20.625z" fill="#8393a7"/><g transform="matrix(.6 0 0 .6875 0 .69)"><clipPath id="a"><path d="M0-1h32.822L8.575 31H0V-1z"/></clipPath><g clip-path="url(#a)"><path d="M0 30c0 .55.45 1 1 1h38c.55 0 1-.45 1-1V6c0-.55-.45-1-1-1H18l-6-6H1C.45-1 0-.55 0 0v30z" fill="#53657d"/></g></g></svg>