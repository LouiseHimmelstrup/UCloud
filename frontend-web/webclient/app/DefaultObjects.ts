import {SidebarOption, Page} from "Types";
import {Status} from "Navigation";
import {Analysis, DetailedApplicationSearchReduxState, RunsSortBy} from "Applications";
import {File, DetailedFileSearchReduxState} from "Files";
import {SortOrder, SortBy} from "Files";
import {DashboardStateProps} from "Dashboard";
import {Publication} from "Zenodo";
import {Notification} from "Notifications";
import {Upload} from "Uploader";
import {Activity, ActivityGroup, ActivityFilter} from "Activity";
import {Reducer} from "redux";
import {SimpleSearchStateProps} from "Search";
import * as ApplicationRedux from "Applications/Redux";
import * as AccountingRedux from "Accounting/Redux";
import * as SnackbarRedux from "Snackbar/Redux";
import * as FavoritesRedux from "Favorites/Redux";
import {defaultAvatar} from "UserSettings/Avataaar";
import {SidebarPages} from "ui-components/Sidebar";
import {ScrollResult} from "Scroll/Types";
import * as ProjectRedux from "Project/Redux";

export const DefaultStatus: Status = {
    title: "No Issues",
    level: "NO ISSUES",
    body: "The system is running as intended."
};

export enum KeyCode {
    ENTER = 13,
    ESC = 27
}

export const emptyPage: Page<any> = {items: [], itemsPerPage: 25, itemsInTotal: 0, pageNumber: 0, pagesInTotal: 0};

export enum AnalysesStatusMap {
    "PENDING",
    "IN PROGRESS",
    "COMPLETED"
}

export enum RightsNameMap {
    "NONE" = "None",
    "READ" = "Read",
    "READ_WRITE" = "Read/Write"
}

export enum SensitivityLevel {
    "INHERIT" = "Inherit",
    "PRIVATE" = "Private",
    "CONFIDENTIAL" = "Confidential",
    "SENSITIVE" = "Sensitive"
}

export type Sensitivity = keyof typeof SensitivityLevel;

export enum SensitivityLevelMap {
    INHERIT = "INHERIT",
    PRIVATE = "PRIVATE",
    CONFIDENTIAL = "CONFIDENTIAL",
    SENSITIVE = "SENSITIVE"
}

function getFilesSortingColumnOrDefault(columnIndex: 0 | 1): SortBy {
    const sortingColumn = window.localStorage.getItem(`filesSorting${columnIndex}`);
    if (sortingColumn && Object.values(SortBy).includes(sortingColumn)) return sortingColumn as SortBy;
    switch (columnIndex) {
        case 0:
            window.localStorage.setItem("filesSorting0", SortBy.MODIFIED_AT);
            return SortBy.MODIFIED_AT;
        case 1:
            window.localStorage.setItem("filesSorting1", SortBy.SIZE);
            return SortBy.SIZE;
    }
}

function getItemOrDefault<T, T2>(itemName: string, defaultValue: T, en: T2): T {
    const item = window.localStorage.getItem(itemName);
    if (item && Object.values(en).includes(item)) {
        return item as unknown as T;
    }
    return defaultValue;
}

export interface ComponentWithLoadingState {
    loading: boolean
    error?: string
}

export interface ComponentWithPage<T> extends ComponentWithLoadingState {
    page: Page<T>
}

export interface ComponentWithScroll<Item, OffsetType> extends ComponentWithLoadingState {
    scroll?: ScrollResult<Item, OffsetType>
}

export interface ResponsiveReduxObject {
    mediaType: string
    orientation: string
    lessThan: Record<string, boolean>
    greaterThan: Record<string, boolean>
    is: Record<string, boolean>
}

export const initResponsive = (): ResponsiveReduxObject => ({
    mediaType: "",
    orientation: "",
    lessThan: {},
    greaterThan: {},
    is: {},
});

export interface FilePreviewReduxState {
    file?: File
    error?: string
}

export interface FilesReduxObject extends ComponentWithPage<File> {
    sortOrder: SortOrder
    sortBy: SortBy
    path: string
    filesInfoPath: string
    fileSelectorError?: string
    sortingColumns: [SortBy, SortBy]
    fileSelectorLoading: boolean
    fileSelectorShown: boolean
    fileSelectorPage: Page<File>
    fileSelectorPath: string
    fileSelectorCallback: Function
    fileSelectorIsFavorites: boolean
    disallowedPaths: string[]
    invalidPath: boolean
}

export interface FileInfoReduxObject {
    file?: File
    error?: string
    activity: Page<Activity>
    loading: boolean
}

export interface AnalysisReduxObject extends ComponentWithPage<Analysis> {
    sortBy: RunsSortBy
    sortOrder: SortOrder
}

export interface NotificationsReduxObject {
    redirectTo: string
    items: Notification[]
    loading: boolean
    error?: string
}

export interface ZenodoReduxObject extends ComponentWithPage<Publication> {
    connected: boolean
}

export interface StatusReduxObject {
    status: Status
    title: string
    page: SidebarPages
    loading: boolean
}

export interface SidebarReduxObject {
    pp: boolean
    options: SidebarOption[]
    kcCount: number
}

export interface HeaderSearchReduxObject {
    prioritizedSearch: HeaderSearchType
    refresh?: () => void
}

export interface RunApplicationReduxObject {

}

export type ActivityReduxObject = ComponentWithScroll<ActivityGroup, number> & ActivityFilter;

export type HeaderSearchType = "files" | "applications" | "projects";

export interface UploaderReduxObject {
    uploads: Upload[]
    visible: boolean
    path: string
    allowMultiple: boolean
    onFilesUploaded: (p: string) => void
    error?: string
    loading: boolean
}

interface LegacyReducers {
    dashboard?: Reducer<DashboardStateProps>
    files?: Reducer<FilesReduxObject>
    uploader?: Reducer<UploaderReduxObject>
    status?: Reducer<StatusReduxObject>
    notifications?: Reducer<NotificationsReduxObject>
    analyses?: Reducer<AnalysisReduxObject>
    zenodo?: Reducer<ZenodoReduxObject>
    header?: Reducer<HeaderSearchReduxObject>
    sidebar?: Reducer<SidebarReduxObject>
    activity?: Reducer<ActivityReduxObject>
    detailedResult?: Reducer<DetailedResultReduxObject>
}

export type Reducers = LegacyReducers & ApplicationRedux.Reducers & AccountingRedux.Reducers;

export type DetailedResultReduxObject = ComponentWithPage<File>

export const initDetailedResult = (): DetailedResultReduxObject => ({
    page: emptyPage,
    loading: false,
    error: undefined
});

/* FIXME */
interface LegacyReduxObject {
    dashboard: DashboardStateProps
    files: FilesReduxObject,
    uploader: UploaderReduxObject
    status: StatusReduxObject,
    notifications: NotificationsReduxObject
    analyses: AnalysisReduxObject
    zenodo: ZenodoReduxObject
    header: HeaderSearchReduxObject
    sidebar: SidebarReduxObject
    activity: ActivityReduxObject
    detailedResult: DetailedResultReduxObject
    simpleSearch: SimpleSearchStateProps
    detailedFileSearch: DetailedFileSearchReduxState
    detailedApplicationSearch: DetailedApplicationSearchReduxState
    fileInfo: FileInfoReduxObject
    avatar: AvatarReduxObject
    filePreview: FilePreviewReduxState
    responsive?: ResponsiveReduxObject
    project: ProjectRedux.State
    loading?: boolean
}

export type ReduxObject =
    LegacyReduxObject &
    ApplicationRedux.Objects &
    AccountingRedux.Objects &
    FavoritesRedux.Objects &
    SnackbarRedux.Wrapper;

export const initActivity = (): ActivityReduxObject => ({
    loading: false
});

export const initNotifications = (): NotificationsReduxObject => ({
    items: [],
    loading: false,
    redirectTo: "",
    error: undefined
});

export const initHeader = (): HeaderSearchReduxObject => ({
    prioritizedSearch: "files"
});

export const initStatus = (): StatusReduxObject => ({
    status: DefaultStatus,
    title: "",
    page: SidebarPages.None,
    loading: false
});

export const initDashboard = (): DashboardStateProps => ({
    favoriteFiles: [],
    recentFiles: [],
    recentAnalyses: [],
    notifications: [],
    favoriteLoading: false,
    recentLoading: false,
    analysesLoading: false
});

export function initObject(homeFolder: string): ReduxObject {
    return {
        dashboard: initDashboard(),
        files: initFiles(homeFolder),
        status: initStatus(),
        header: initHeader(),
        notifications: initNotifications(),
        analyses: initAnalyses(),
        zenodo: initZenodo(),
        sidebar: initSidebar(),
        uploader: initUploads(),
        activity: initActivity(),
        detailedResult: initDetailedResult(),
        simpleSearch: initSimpleSearch(),
        detailedApplicationSearch: initApplicationsAdvancedSearch(),
        detailedFileSearch: initFilesDetailedSearch(),
        fileInfo: initFileInfo(),
        avatar: initAvatar(),
        filePreview: initFilePreview(),
        project: ProjectRedux.initialState,
        ...ApplicationRedux.init(),
        ...AccountingRedux.init(),
        ...FavoritesRedux.init(),
        ...SnackbarRedux.init(),
        responsive: undefined,
    }
}


export const initFilePreview = () => ({
    file: undefined,
    error: undefined
});

export type AvatarReduxObject = typeof defaultAvatar & { error?: string };
export const initAvatar = (): AvatarReduxObject => ({...defaultAvatar, error: undefined});

export const initSimpleSearch = (): SimpleSearchStateProps => ({
    files: emptyPage,
    filesLoading: false,
    applications: emptyPage,
    applicationsLoading: false,
    errors: [],
    search: "",
    applicationSearch: initApplicationsAdvancedSearch(),
    fileSearch: initFilesDetailedSearch()
});

export const initAnalyses = (): AnalysisReduxObject => ({
    page: emptyPage,
    loading: false,
    error: undefined,
    sortBy: RunsSortBy.createdAt,
    sortOrder: SortOrder.ASCENDING
});

export const initZenodo = (): ZenodoReduxObject => ({
    connected: false,
    loading: false,
    page: emptyPage
});

export const initSidebar = (): SidebarReduxObject => ({
    pp: false,
    kcCount: 0,
    options: []
});

export const initUploads = (): UploaderReduxObject => ({
    path: "",
    uploads: [],
    visible: false,
    allowMultiple: false,
    error: undefined,
    onFilesUploaded: () => null,
    loading: false
});

export const initFileInfo = (): FileInfoReduxObject => ({
    activity: emptyPage,
    loading: false
});

export const initFiles = (homeFolder: string): FilesReduxObject => ({
    page: emptyPage,
    sortOrder: getItemOrDefault("sortOrder", SortOrder.ASCENDING, SortOrder),
    sortBy: getItemOrDefault("sortBy", SortBy.PATH, SortBy),
    loading: false,
    error: undefined,
    path: "",
    filesInfoPath: "",
    sortingColumns: [getFilesSortingColumnOrDefault(0), getFilesSortingColumnOrDefault(1)],
    fileSelectorLoading: false,
    fileSelectorShown: false,
    fileSelectorPage: emptyPage,
    fileSelectorPath: homeFolder,
    fileSelectorIsFavorites: false,
    fileSelectorCallback: () => undefined,
    fileSelectorError: undefined,
    disallowedPaths: [],
    invalidPath: false
});

export const initFilesDetailedSearch = (): DetailedFileSearchReduxState => ({
    hidden: true,
    allowFolders: true,
    allowFiles: true,
    fileName: "",
    extensions: new Set(),
    tags: new Set(),
    sensitivities: new Set(),
    createdBefore: undefined,
    createdAfter: undefined,
    modifiedBefore: undefined,
    modifiedAfter: undefined,
    error: undefined,
    loading: false
});

export const initApplicationsAdvancedSearch = (): DetailedApplicationSearchReduxState => ({
    error: undefined,
    loading: false,
    hidden: true,
    appName: "",
    appVersion: "",
    tags: ""
});
