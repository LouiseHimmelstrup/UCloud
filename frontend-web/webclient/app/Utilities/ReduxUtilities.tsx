import dashboard from "Dashboard/Redux/DashboardReducer";
import {initObject} from "DefaultObjects";
import header, {CONTEXT_SWITCH, USER_LOGIN, USER_LOGOUT} from "Navigation/Redux/HeaderReducer";
import sidebar from "Navigation/Redux/SidebarReducer";
import status from "Navigation/Redux/StatusReducer";
import notifications from "Notifications/Redux/NotificationsReducer";
import * as ProjectRedux from "Project/Redux";
import {Action, AnyAction, combineReducers, createStore, Store} from "redux";
import {composeWithDevTools} from "redux-devtools-extension";
import {createResponsiveStateReducer, responsiveStoreEnhancer} from "redux-responsive";
import simpleSearch from "Search/Redux/SearchReducer";
import {responsiveBP} from "ui-components/theme";
import avatar from "UserSettings/Redux/AvataaarReducer";
import hookStore from "Utilities/ReduxHooks";

export function configureStore(
    initialObject: ReduxObject,
    reducers,
    enhancers?
): Store<ReduxObject, AnyAction> {
    const combinedReducers = combineReducers<ReduxObject, AnyAction>(reducers);
    const rootReducer = (state: ReduxObject, action: Action): ReduxObject => {
        if ([USER_LOGIN, USER_LOGOUT, CONTEXT_SWITCH].some(it => it === action.type)) {
            state = initObject();
        }
        return combinedReducers(state, action);
    };
    return createStore<ReduxObject, AnyAction, {}, {}>(rootReducer, initialObject, composeWithDevTools(enhancers));
}

export const responsive = createResponsiveStateReducer(
    responsiveBP,
    {infinity: "xxl"}
);

export const store = configureStore(initObject(), {
    dashboard,
    header,
    status,
    sidebar,
    notifications,
    simpleSearch,
    hookStore,
    avatar,
    loading,
    project: ProjectRedux.reducer,
    responsive: createResponsiveStateReducer(
        responsiveBP,
        {infinity: "xxl"}),
}, responsiveStoreEnhancer);

function loading(state = false, action: {type: string}): boolean {
    switch (action.type) {
        case "LOADING_START":
            return true;
        case "LOADING_END":
            return false;
        default:
            return state;
    }
}
