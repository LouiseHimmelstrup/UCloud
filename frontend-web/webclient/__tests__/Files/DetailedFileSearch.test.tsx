import * as React from "react";
import * as Renderer from "react-test-renderer";
import DetailedFileSearch from "Files/DetailedFileSearch";
import { Provider } from "react-redux";
import { configureStore } from "Utilities/ReduxUtilities";
import { initFiles } from "DefaultObjects";
import files from "Files/Redux/FilesReducer";
import { mount, configure } from "enzyme";
import * as Adapter from "enzyme-adapter-react-16";
import * as moment from "moment";
import { DatePicker } from "ui-components/DatePicker"
import "react-datepicker/dist/react-datepicker.css";
import { Input, Button, Label } from "ui-components";

configure({ adapter: new Adapter() });

const store = configureStore({ files: initFiles("/home/person@place.tv") }, { files });

describe("DetailedFileSearch", () => {
    test.skip("Detailed File Search component", () => {
        expect(Renderer.create(
            <Provider store={store}>
                <DetailedFileSearch />
            </Provider>
        ).toJSON()).toMatchSnapshot();
    });

    test.skip("Add filename", () => {
        const filename = "nom de file";
        const detailedFileSearchWrapper = mount(
            <Provider store={store}>
                <DetailedFileSearch />
            </Provider>
        );
        detailedFileSearchWrapper.find(Button).simulate("click");
        detailedFileSearchWrapper.findWhere(it => it.props().placeholder === "Filename must include...").find("input").simulate("change", { target: { value: filename } });
        expect(detailedFileSearchWrapper.find(DetailedFileSearch).childAt(0).state("fileName")).toBe(filename);
    });

    test.skip("Add dates to fields", () => {
        const m = moment();
        const detailedFileSearchWrapper = mount(
            <Provider store={store}>
                <DetailedFileSearch />
            </Provider>
        );
        detailedFileSearchWrapper.find(Button).simulate("click");
        detailedFileSearchWrapper.find(DatePicker).forEach(it => it.find("input").simulate("change", { target: { value: m } }));
        expect((detailedFileSearchWrapper.find(DetailedFileSearch).childAt(0).state("createdBefore"))).toEqual(m);
        expect((detailedFileSearchWrapper.find(DetailedFileSearch).childAt(0).state("createdAfter"))).toEqual(m);
        expect((detailedFileSearchWrapper.find(DetailedFileSearch).childAt(0).state("modifiedBefore"))).toEqual(m);
        expect((detailedFileSearchWrapper.find(DetailedFileSearch).childAt(0).state("modifiedAfter"))).toEqual(m);
    });

    test.skip("Add date, causing one field to disappear, and render an error message", () => {
        const m1 = moment(new Date());
        const m2 = moment(new Date(new Date().getMilliseconds() - 500));
        const detailedFileSearchWrapper = mount(
            <Provider store={store}>
                <DetailedFileSearch />
            </Provider>
        );
        detailedFileSearchWrapper.find(Button).simulate("click");
        detailedFileSearchWrapper.find(DatePicker).first().find("input").simulate("change", { target: { value: m1 } });
        expect((detailedFileSearchWrapper.find(DetailedFileSearch).childAt(0).instance().state as any).createdAfter).toBeDefined();
        detailedFileSearchWrapper.find(DatePicker).slice(1, 2).find("input").simulate("change", { target: { value: m2 } });
        expect((detailedFileSearchWrapper.find(DetailedFileSearch).childAt(0).instance().state as any).createdAfter).toBeUndefined();
        // FIXME When error messages are better handled for detailedFileSearch, dismiss error;
    });

    test.skip("Deselect folder and file checkboxes", () => {
        const detailedFileSearch = mount(
            <Provider store={store}>
                <DetailedFileSearch />
            </Provider>
        );
        detailedFileSearch.find(Button).simulate("click");
        console.log(detailedFileSearch.find(Label).length);
        const folderCheckbox = detailedFileSearch.find(Label).first();
        const filesCheckbox = detailedFileSearch.find(Label).last();
        expect((detailedFileSearch.find(DetailedFileSearch).childAt(0).state()).allowFolders).toBe(true);
        expect((detailedFileSearch.find(DetailedFileSearch).childAt(0).state()).allowFiles).toBe(true);
        filesCheckbox.simulate("click", { target: { checked: false } });
        expect((detailedFileSearch.find(DetailedFileSearch).childAt(0).state()).allowFolders).toBe(true);
        expect((detailedFileSearch.find(DetailedFileSearch).childAt(0).state()).allowFiles).toBe(false);
        expect((detailedFileSearch.find(DetailedFileSearch).childAt(0).state()).allowFolders).toBe(false);
        expect((detailedFileSearch.find(DetailedFileSearch).childAt(0).state()).allowFiles).toBe(false);
        filesCheckbox.find("input").simulate("click");
        folderCheckbox.find("input").simulate("click");
        expect((detailedFileSearch.find(DetailedFileSearch).childAt(0).state()).allowFolders).toBe(true);
        expect((detailedFileSearch.find(DetailedFileSearch).childAt(0).state()).allowFiles).toBe(true);
    });

    test.skip("Add sensitivities, clear one, clear all", () => {
        const detailedFileSearchWrapper = mount(
            <Provider store={store}>
                <DetailedFileSearch />
            </Provider>
        );
        detailedFileSearchWrapper.find(Button).simulate("click");
        const sensitivityDropdown = detailedFileSearchWrapper.find("Dropdown").findWhere(it => it.props().text === "Add sensitivity level");
        const detailedFileSearchComponent = detailedFileSearchWrapper.find(DetailedFileSearch).childAt(0);
        expect((detailedFileSearchComponent.instance().state as any).sensitivities.has("Open Access")).toBe(false);
        sensitivityDropdown.findWhere(it => it.props().text === "Open Access").simulate("click");
        expect((detailedFileSearchComponent.instance().state as any).sensitivities.has("Open Access")).toBe(true);
        detailedFileSearchWrapper.find(Label).children().first().children().simulate("click");
        expect((detailedFileSearchComponent.instance().state as any).sensitivities.has("Open Access")).toBe(false);
        sensitivityDropdown.findWhere(it => it.props().text === "Sensitive").simulate("click");
        sensitivityDropdown.findWhere(it => it.props().text === "Confidential").simulate("click");
        expect((detailedFileSearchComponent.instance().state as any).sensitivities.size).toBe(2);
        detailedFileSearchWrapper.find(Label).children().last().children().first().simulate("click");
        expect((detailedFileSearchComponent.instance().state as any).sensitivities.size).toBe(0);

    });

    test.skip("Add extensions, clear one, clear all", () => {
        const extensions = ".ext1 .ext2";
        const detailedFileSearchWrapper = mount(
            <Provider store={store}>
                <DetailedFileSearch />
            </Provider>
        );
        detailedFileSearchWrapper.find(Button).simulate("click");
        detailedFileSearchWrapper.find(Input).findWhere(it => it.props().placeholder === "Add extensions...").find("input").simulate("change", { target: { value: extensions } })
        const extensionDropdown = detailedFileSearchWrapper.find("Dropdown").findWhere(it => it.props().text === "Add extension preset");
        // FIXME

    });

    test.skip("Add extensions from presets, clear one, clear all", () => {
        // FIXME
    });

    test.skip("Add tag filename, clear one, clear all", () => {
        // FIXME
    });
});