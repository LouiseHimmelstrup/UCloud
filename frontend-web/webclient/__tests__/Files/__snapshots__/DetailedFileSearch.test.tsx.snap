// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailedFileSearch Detailed File Search component 1`] = `
<div
  className="ui container sixteen column grid"
>
  <div
    className="sixteen wide column"
  >
    <h3
      className="ui header"
    >
      Filename
    </h3>
    <div
      className="ui fluid input"
    >
      <input
        onChange={[Function]}
        placeholder="Filename must include..."
        type="text"
      />
    </div>
    <h3
      className="ui header"
    >
      Created at
    </h3>
    <form
      className="ui form"
      onSubmit={[Function]}
    >
      <div
        className="fields"
      >
        <div
          className="field"
        >
          <label>
            Created after
          </label>
          <div>
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  className=""
                  disabled={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  readOnly={false}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="field"
        >
          <label>
            Created before
          </label>
          <div>
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  className=""
                  disabled={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  readOnly={false}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
    <h3
      className="ui header"
    >
      Modified at
    </h3>
    <form
      className="ui form"
      onSubmit={[Function]}
    >
      <div
        className="fields"
      >
        <div
          className="field"
        >
          <label>
            Modified after
          </label>
          <div>
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  className=""
                  disabled={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  readOnly={false}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
        <div
          className="field"
        >
          <label>
            Modified before
          </label>
          <div>
            <div
              className="react-datepicker-wrapper"
            >
              <div
                className="react-datepicker__input-container"
              >
                <input
                  className=""
                  disabled={false}
                  onBlur={[Function]}
                  onChange={[Function]}
                  onClick={[Function]}
                  onFocus={[Function]}
                  onKeyDown={[Function]}
                  readOnly={false}
                  type="text"
                  value=""
                />
              </div>
            </div>
          </div>
        </div>
      </div>
    </form>
    <h3
      className="ui header"
    >
      File Types
    </h3>
    <div
      className="inline fields"
    >
      <div
        className="field"
      >
        <div
          className="ui checked checkbox"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={true}
            className="hidden"
            onClick={[Function]}
            readOnly={true}
            tabIndex={0}
            type="checkbox"
          />
          <label>
            Folders
          </label>
        </div>
      </div>
      <div
        className="field"
      >
        <div
          className="ui checked checkbox"
          onChange={[Function]}
          onClick={[Function]}
          onMouseDown={[Function]}
        >
          <input
            checked={true}
            className="hidden"
            onClick={[Function]}
            readOnly={true}
            tabIndex={0}
            type="checkbox"
          />
          <label>
            Files
          </label>
        </div>
      </div>
    </div>
    <h3
      className="ui header"
    >
      File extensions
    </h3>
    <div
      className="padding-bottom"
    />
    <form
      className="ui form"
      onSubmit={[Function]}
    >
      <div
        className="field"
      >
        <div
          className="ui input"
        >
          <input
            onChange={[Function]}
            placeholder="Add extensions..."
            type="text"
            value=""
          />
        </div>
      </div>
      <div
        aria-expanded={false}
        className="ui dropdown"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        role="listbox"
        tabIndex={0}
      >
        <div
          aria-live="polite"
          className="text"
          role="alert"
        >
          Add extension preset
        </div>
        <i
          aria-hidden="true"
          className="dropdown icon"
          onClick={[Function]}
        />
        <div
          className="menu transition"
        >
          <div
            aria-checked={false}
            aria-selected={true}
            className="selected item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              Text
            </span>
          </div>
          <div
            aria-checked={false}
            aria-selected={false}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              Image
            </span>
          </div>
          <div
            aria-checked={false}
            aria-selected={false}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              Sound
            </span>
          </div>
          <div
            aria-checked={false}
            aria-selected={false}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              Compressed files
            </span>
          </div>
        </div>
      </div>
    </form>
    <h3
      className="ui header"
    >
      Sensitivity
    </h3>
    <div
      className="padding-bottom"
    />
    <div>
      <div
        aria-expanded={false}
        className="ui dropdown"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        role="listbox"
        tabIndex={0}
      >
        <div
          aria-live="polite"
          className="text"
          role="alert"
        >
          Add sensitivity level
        </div>
        <i
          aria-hidden="true"
          className="dropdown icon"
          onClick={[Function]}
        />
        <div
          className="menu transition"
        >
          <div
            aria-checked={false}
            aria-selected={true}
            className="selected item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              Open Access
            </span>
          </div>
          <div
            aria-checked={false}
            aria-selected={false}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              Confidential
            </span>
          </div>
          <div
            aria-checked={false}
            aria-selected={false}
            className="item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              Sensitive
            </span>
          </div>
        </div>
      </div>
    </div>
    <h3
      className="ui header"
    >
      Annotations
    </h3>
    <div
      className="padding-bottom"
    />
    <div>
      <div
        aria-expanded={false}
        className="ui dropdown"
        onBlur={[Function]}
        onChange={[Function]}
        onClick={[Function]}
        onFocus={[Function]}
        onMouseDown={[Function]}
        role="listbox"
        tabIndex={0}
      >
        <div
          aria-live="polite"
          className="text"
          role="alert"
        >
          Add annotation
        </div>
        <i
          aria-hidden="true"
          className="dropdown icon"
          onClick={[Function]}
        />
        <div
          className="menu transition"
        >
          <div
            aria-checked={false}
            aria-selected={true}
            className="selected item"
            onClick={[Function]}
            role="option"
            style={
              Object {
                "pointerEvents": "all",
              }
            }
          >
            <span
              className="text"
            >
              Project
            </span>
          </div>
        </div>
      </div>
    </div>
    <h3
      className="ui header"
    >
      Tags
    </h3>
    <div
      className="padding-bottom"
    />
    <form
      className="ui form"
      onSubmit={[Function]}
    >
      <div
        className="field"
      >
        <div
          className="ui input"
        >
          <input
            onChange={[Function]}
            type="text"
            value=""
          />
        </div>
      </div>
    </form>
    <button
      className="ui blue button"
      onClick={[Function]}
      role="button"
      style={
        Object {
          "marginTop": "15px",
        }
      }
    >
      Search
    </button>
  </div>
</div>
`;
